src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,142 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_ARRAY_SIZE = 10;
+    private T[] array;
+    private int arraySize = 0;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_ARRAY_SIZE];
+    }
+
+    private void resize() {
```

Comment:
The `resize` method is doing more than just resizing. It's also copying the array. Consider renaming it to reflect this, such as 'resizeAndCopy'.